text/0015-interest-management.md

```diff
@@ -0,0 +1,261 @@
+
+But now let's say the network id of objects *can* change, however it's infrequent.  What's more, the game code knows when this happens to which object.  We can implement that simply with the `UpdateObject` method
+
+```
+        public override void UpdateObject(in NetworkObject obj)
+        {
+            RemoveObject(obj);
+            AddObject(obj);
+        }
+```
+
+Here, the game code can call into the node itself and pass in the changed object.  Here we chose to simply remove it from itself and re-add it, letting the standard processing to determine which bucket it goes in take place.
+
+With this in mind we can think about the analogues in the more realistic graph node example
+
+* in the same way, objects move around from cell to cell.  However, we don't want the Interest Node to have to re-scan all the objects every frame to re-assign them to the proper graph.  We assume the user code will likely know more about when these things happen (e.g. an object teleports instantly).  OR, we imagine the developer will want to strategize; if a player is subscribed as is common to a cluster of overlapping cells with objects sometimes belonging to > 1 cell, then - assuming max object velocity is taken into account - one can safely re-scan objects into cells much, much less often then every frame
+* Developers have the option to avoid every re-scanning some objects at all.  Say we have a 'tree' prefab and an associated grid node associated with it.  The developer could:
+  * make a special `InterestedGraphNodeDormant` or other that has no `UpdateObject` body, and/or
+  * in its own code simply know what objects never need to be updated and not call `UpdateObject` on them.
+
+
+# FAQ
+
+## Why can one associate more than one `InterestNode` with a prefab?
+
+Let's imagine you're setting up your player's prefab.  Say you want players can see other players if they're some distance away using a dispatched / custom storage like the second example.  But let's say we also want to enable players to see other players who are on the same team, regardless of distance.  You could write a special `RadiusOrTeam` combination node that does both.  For better re-use, however, the system allows one to add more than one node; in this case you would have a `GraphInterestNode` *and* a `TeamInterestNode`.  Then, when the query happens on player prefabs the result from both nodes is merged
+
+
+
+# Drawbacks
+- The U/X is a bit cumbersome in the `InterestNodeStatic` case in that you must create a node, then create the kernel and then associate both to the prefab
+
+# Prior art
+
+# Unresolved questions
+- as an alternative to having storage of nodes under management in `InterestNodeStatic`, what about an API change to allow a node to simply get access to all spawned objects?
+- or, put another way, should the Interest System become the single source of truth for `SpawnManager.SpawnedObjects`?
```

Comment:
This would make sense.  This way you could have a default radius for your query of spawned objects, but if needed you might need to perform predictive calculations that requires viewing "beyond your current" AOI radius.  This would be useful for late joining player synchronization (near future) and streaming worlds (etc).